sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -214,23 +221,31 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
     uncacheByCondition(spark, _.sameResult(plan), cascade, blocking)
   }
 
-  def uncacheTableOrView(spark: SparkSession, name: Seq[String], cascade: Boolean): Unit = {
+  def uncacheTableOrView(
+      spark: SparkSession,
+      name: Seq[String],
+      cascade: Boolean,
+      blocking: Boolean = false): Unit = {
     uncacheByCondition(
-      spark, isMatchedTableOrView(_, name, spark.sessionState.conf), cascade, blocking = false)
+      spark, isMatchedTableOrView(_, name, spark.sessionState.conf), cascade, blocking)
   }
 
-  private def isMatchedTableOrView(plan: LogicalPlan, name: Seq[String], conf: SQLConf): Boolean = {
+  private def isMatchedTableOrView(
+      plan: LogicalPlan,
+      name: Seq[String],
+      conf: SQLConf,
+      includeTimeTravel: Boolean = true): Boolean = {
     def isSameName(nameInCache: Seq[String]): Boolean = {
       nameInCache.length == name.length && nameInCache.zip(name).forall(conf.resolver.tupled)
     }
 
-    plan match {
+    EliminateSubqueryAliases(plan) match {
```

Comment:
This change is needed as our invalidation by name without cascading is not working.

```
cacheManager.uncacheTableOrView(spark, Seq("testcat", "tbl"), cascade = false)
``` 

Would not find the table in `CacheManager` as it is wrapped into a subquery alias.

Reply:
Good catch! It this a bug without your PR as well?